openstack/baremetal/v1/nodes/results.go

```diff
@@ -50,6 +50,9 @@ func ExtractNodesInto(r pagination.Page, v interface{}) error {
 
 // Node represents a node in the OpenStack Bare Metal API.
 type Node struct {
+	// Whether automated cleaning is enabled or disabled on this node.
+	AutomatedClean bool `json:"automated_clean"`
```

Comment:
The field will be missing on old microversions, you need to avoid defaulting to `false`.

Reply:
@dtantsur 
I'm not sure how to enforce the default value here, but maybe we could have
```
AutomatedClean *bool `json:"automated_clean"
```
and set it to `true` if we see nil value? I assume this would need to be done in Ironic code :(

Reply:
If you use `*bool` as the type then `omitempty` should work (`nil` = empty if I understand Go right).

Reply:
Oh, wait, my comment was about a request. I'm not sure, I guess we can live with `nil` if ironic doesn't return this field? You need to ask someone who knows more about Go, to be honest.

Reply:
If `AutomatedClean` is a field that was added from a microversion, then please use `*bool`. This way the app/developer can check if it is `nil` or not. It's a little hacky, but the most sane solution I could come up with.

Also, please include a docstring that details what microversion the field was added in.

And let me know if anyone has any questions :) 

https://github.com/gophercloud/gophercloud/blob/master/docs/MICROVERSIONS.md#new-response-fields

Reply:
Thank you @dtantsur and @jtopjian . Fixed.